crates/c-api/include/wasmtime/store.h

```diff
@@ -243,6 +253,45 @@ WASM_API_EXTERN void wasmtime_context_set_epoch_deadline(wasmtime_context_t *con
  */
 WASM_API_EXTERN void wasmtime_store_epoch_deadline_callback(wasmtime_store_t *store, wasmtime_error_t* (*func)(wasmtime_context_t*, void*, uint64_t*), void *data);
 
+/**
+ * \brief Insert a file descriptor into the underlying WASI context. 
+ * 
+ * This is used to dynamically bind a file descriptor from the host to a file 
+ * descriptor in an already-built WASI context to make it available to WASI APIs.
+ * And it requires that the WASI context has been configured correctly prior to
+ * calling this function.
+ * 
+ * The `guest_fd` argument is the number of the file descriptor by which it will
+ * be known in WASM and the `host_fd` is the number of the file descriptor on
+ * the host and the `access_mode` is the access mode of the file descriptor. 
+ * The enumeration of access modes aligns with FileAccessMode in the WASI spec 
+ * and as follows: READ = 0b1, WRITE = 0b10.
```

Comment:
I think the documentation here and below may wish to be updated where these constants are not defined by the WASI spec, they're purely an implementation detail in Wasmtime.

Could this additionally split out the read/write to `WASMTIME_WASI_FILE_{READ,WRITE}` perhaps as `#define`s?

Reply:
> not defined by the WASI spec

Then possibly some better documentation is needed on the wasi_common package instead. I found no indication saying this is a pure wasmtime thing.